public/app/features/alerting/unified/hooks/useAlertmanagerConfig.ts

```diff
@@ -1,26 +1,23 @@
+import { SerializedError } from '@reduxjs/toolkit';
 import { useEffect } from 'react';
 
-import { useDispatch } from 'app/types';
-
-import { fetchAlertManagerConfigAction } from '../state/actions';
-import { initialAsyncRequestState } from '../utils/redux';
-
-import { useUnifiedAlertingSelector } from './useUnifiedAlertingSelector';
+import { alertmanagerApi } from '../api/alertmanagerApi';
 
+// TODO refactor this so we can just call "alertmanagerApi.endpoints.getAlertmanagerConfiguration" everywhere
+// and remove this hook
 export function useAlertmanagerConfig(amSourceName?: string) {
-  const dispatch = useDispatch();
+  const [fetchConfig, fetchState] = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useLazyQuery();
 
   useEffect(() => {
     if (amSourceName) {
-      dispatch(fetchAlertManagerConfigAction(amSourceName));
+      fetchConfig(amSourceName);
     }
-  }, [amSourceName, dispatch]);
-
-  const amConfigs = useUnifiedAlertingSelector((state) => state.amConfigs);
-
-  const { result, loading, error } = (amSourceName && amConfigs[amSourceName]) || initialAsyncRequestState;
-
-  const config = result?.alertmanager_config;
-
-  return { result, config, loading, error };
+  }, [amSourceName, fetchConfig]);
+
+  return {
+    result: fetchState.data,
+    config: fetchState.data?.alertmanager_config,
+    loading: fetchState.isLoading,
+    error: fetchState.error as SerializedError,
```

Comment:
I really don't know any other way to type this since RTK Query will _always_ mark it as `unknown`, which is technically true. 

The only reason I've typed it here as `SerializedError` is because I've manually checked all of the code paths in `getAlertmanagerConfiguration` to return `withSerializedError()`, which is not great.

I'll probably send some more PRs to get rid of this type assertion.

Reply:
maybe we could add a todo here?

Reply:
Could `isFetchError` help here?

Reply:
No not really since `FetchError` is different from `SerializedError` :(